homeassistant/components/sensor/__init__.py

```diff
@@ -302,36 +302,62 @@ def state(self) -> Any:
             # Anyways, lets validate the date at least..
             try:
                 value = ciso8601.parse_datetime(str(value))
-            except (ValueError, IndexError) as error:
-                raise ValueError(
-                    f"Invalid date/datetime: {self.entity_id} provide state '{value}', "
-                    f"while it has device class '{device_class}'"
-                ) from error
+            except (ValueError, IndexError):
+                _LOGGER.error(
+                    "Invalid date/datetime: %s provide state '%s', "
+                    "while it has device class '%s'",
+                    self.entity_id,
+                    value,
+                    device_class,
+                )
+                return None
 
             # Convert the date object to a standardized state string.
             if device_class == DEVICE_CLASS_DATE:
                 return value.date().isoformat()
-            return value.isoformat(timespec="seconds")
+
+            if value.tzinfo != timezone.utc:
+                value = value.astimezone(timezone.utc)
+
+            return value.astimezone(timezone.utc).isoformat(timespec="seconds")
```

Comment:
Is the check needed, or is this enough?
```suggestion
            return value.astimezone(timezone.utc).isoformat(timespec="seconds")
```

Reply:
This is actually not good, as we should (for backward compatibility) accept strings without timezone in the deprecation handling.

This suggestion exposes a bug! Thanks 👍

Reply:
Fixed in 014b49c, added tests in there for that case as well.